docs/useragents.rst

```diff
@@ -0,0 +1,210 @@
+Mandatory User-Agent
+====================
+
+This is a guide on how to implement *Kiwi.com RFC #22: Mandatory User Agents
+for service-to-service communication* in Python applications and libraries.
+
+The RFC requires that all HTTP requests made to an internally developed
+service must include a ``User-Agent`` header in the following format::
+
+    <service name>/<version> (Kiwi.com <environment>) [<system info>]
+
+In order to achieve this, the RFC also requires that all internal services
+must refuse requests which do not comply with this format. Essentially,
+the following kinds of changes must be made in all applications and libraries:
+
+1. all HTTP requests made to an internally developed service must include a
+   ``User-Agent`` header complying with the KW-RFC-22 format
+2. all internally developed services must refuse requests from clients which
+   do not comply with the KW-RFC-22 format
+
+The ``kiwi-platform`` library helps to resolve both requirements by
+providing:
+
+1. `Making HTTP requests`_ - custom client sessions for making HTTP requests
+   and a patching mechanism
+2. `Validating client requests`_ - custom middlewares for validating
+   client's ``User-Agent`` header
+
+Making HTTP requests
+--------------------
+
+The library provides custom sessions for two most used 3rd party libraries:
+
+- `Python-Requests <https://2.python-requests.org/en/master/>`_
+- `AIOHTTP <https://aiohttp.readthedocs.io/en/stable/>`_
+
+In order to use them, your application need to populate the following
+environment variables:
+
+- ``APP_NAME`` - name of the application
+- ``PACKAGE_VERSION`` - either the git commit hash or a version number,
+  for example, ``git-a4f93`` or ``1.0.3``
+- ``APP_ENVIRONMENT`` - a string that matches the environment reported,
+  for example, to Datadog and Sentry, in most cases this would mean
+  ``production``
+
+If you have the variables properly configured, you can use the
+:class:`kw.platform.requests.session.KiwiSession` session which automatically
+adds KW-RFC-22 compliant User-Agent to all HTTP requests::
+
+    from kw.platform.requests import KiwiSession
+
+    session = KiwiSession()
+    session.get("https://api.example.com")
+
+For async applications using AIOHTTP, you can use the
+:class:`kw.platform.aiohttp.session.KiwiClientSession` session::
+
+    from kw.platform.aiohttp import KiwiClientSession
+
+    async with aiohttp.KiwiClientSession() as client:
+        async with client.get("https://api.example.com") as resp:
+            html = await resp.text()
+            print(html)
+
+In case you are not using :mod:`requests` or :mod:`aiohttp`, you can still use
```

Comment:
What will these `:mod:` references point to? If it's something sensible, why not always refer to the libs like this?